File: back/src/main/java/com/tdevilleduc/urthehero/back/model/Page.java
package com.tdevilleduc.urthehero.back.model;

import lombok.*;

import javax.persistence.*;
import java.util.Collections;
import java.util.List;

@Entity
@NoArgsConstructor
@RequiredArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class Page {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NonNull
    private Integer id;
    @NonNull
    private String text;
    @NonNull
    private String image;

    @Transient
    private List<NextPage> nextPageList = Collections.emptyList();

}

File: back/src/main/java/com/tdevilleduc/urthehero/back/model/Page.kt
package com.tdevilleduc.urthehero.back.model

import io.swagger.v3.oas.annotations.media.Schema
import javax.persistence.*

@Entity
class Page(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Schema(description = "\${swagger.model.page.param.storyId}")
        val id: Int,
        var text: String = "",
        var image: String = "",
        var storyId: Int? = null
) {

    @Transient
    var nextPageList: MutableList<NextPage> = emptyList<NextPage>().toMutableList()

}